### ISSUE TITLE
Create a sentieon specific subworkflow for DSL2

### ISSUE THREAD

**Issue by maxulysse** [nf-core/sarek#405]
EDIT (from @cjfields): 
Modules checklist:

- [x] alignment (accelerated bwa)
- [ ] ~BQSR (optional?) (EDIT: Won't be implemented. It should be possible to use GATK-based BQSR.)~
- [x] DNAseq (SNV/Indel) - including GVCF support
- [x] DNAscope (SV and SNV/Indel) - including GVCF support
- [ ] TNseq (tumor/normal) - including GVCF support
- [x] TNscope (tumor/normal)

**Comment by cjfields** [comment]
Checklist (some are already in place)

**Modules**

- [ ] alignment (accelerated bwa)
- [ ] BQSR (optional?)
- [ ] DNAseq (SNV/Indel) - including GVCF support
- [ ] DNAscope (SV and SNV/Indel)
- [ ] TNseq (tumor/normal) - including GVCF support
- [ ] TNscope (tumor/normal)

**Comment by FriederikeHanssen** [comment]
BQSR would be done also with a Sentieon subtool or GATK?

**Comment by maxulysse** [comment]
Sentieon has that if I remember well

**Comment by cjfields** [comment]
> Sentieon has that if I remember well

Correct; it also supports VQSR though I'm not sure if that's on the roadmap here, I've been running it separately.

**Comment by FriederikeHanssen** [comment]
carefully toying around with it, is the correct answer I would say.

**Comment by nicorap** [comment]
we have a few modules ready from the danish national genome center, I'll just a add them and make the missing ones also with a sub workflow.

**Comment by cjfields** [comment]
Small note: in the 2.7.2 workflow we found a bug which skips Sentieon DNAscope and DNAseq when using manually recalibrated BAMs, which seems to be due to a difference in how GATK and Sentieon steps are implemented.  GATK HaplotypeCaller, Manta, etc) appear to use a recalibrated BAM file with index [here](https://github.com/nf-core/sarek/blob/e8f56e5bbb6f8a34793a6b5a2945981eb43090aa/main.nf#L1831).  

However, the [next line of code](https://github.com/nf-core/sarek/blob/e8f56e5bbb6f8a34793a6b5a2945981eb43090aa/main.nf#L1833) indicates Sentieon steps are using the _deduped_ BAM + index + recalibration table, and the related processes require that table.  So, when given a recalibrated BAM + index, this effectively skips Sentieon calling altogether.  The input channel for those steps also isn't generated when given a table like [this](https://nf-co.re/sarek/2.7.2/usage#input-lttsvgt---step-recalibrate).  

The reason I point this out: we can add a simple fix for this and test it, but it's worth noting that all others steps apart from Sentieon use an already recalibrated BAM file and the Sentieon BQSR generates this BAM file anyway, so I'm not sure of the benefit of having this oddly unique step just for Sentieon.  Maybe something to consider for the DSL2 implementation?

**Comment by nicorap** [comment]
That makes sense, and also explain why I never got to have sentieon running using 2.7.2 from bam. 😐. I think it’s important to be able to skip BQSR, as for production setups the data should always be the same, so it’s not needed.

**Comment by maxulysse** [comment]
@asp8200 I feel we're getting close to finalize this issue

**Comment by asp8200** [comment]
TNseq and  TNscope still not implemented :-/

**Comment by maxulysse** [comment]
yeah I know, but we're getting closer and closer to completion

**Comment by cjfields** [comment]
@maxulysse you are awesome for taking this one, we're about to get this running for some of our projects.

**Comment by maxulysse** [comment]
@asp8200 did all the work, he should get the praise.
You should be able to test it, it's in the latest release already
